### PR TITLE
[net7.0] Cache the dynamic AppTheme value in Controls (and not Essentials) (#11200)

### PR DESCRIPTION
**Pull request by mattleibow**
Backport of #11200 to net7.0

/cc @mattleibow @jonathanpeppers @jsuarezruiz  @PureWeen @Redth

### CODE REVIEW

**Review by jsuarezruiz — APPROVED** [review]
Passing the Build. LGTM.

File: src/Controls/tests/Core.UnitTests/MockPlatformServices.cs [hunk 1]
```diff
@@ -84,6 +84,9 @@ public class MockApplication : Application
 		public MockApplication()
 		{
 		}
+
+		public void NotifyThemeChanged() =>
+			(this as IApplication).ThemeChanged();
```
Comment (mandel-macaque) [inline src/Controls/tests/Core.UnitTests/MockPlatformServices.cs:89]: Do we really need the 'as'?

File: src/Essentials/src/AppInfo/AppInfo.android.cs [hunk 2]
```diff
@@ -44,14 +42,21 @@ public void ShowSettingsUI()
...
 
-		static AppTheme GetRequestedTheme() => (Application.Context.Resources.Configuration.UiMode & UiMode.NightMask) switch
+		static AppTheme GetRequestedTheme()
 		{
-			UiMode.NightYes => AppTheme.Dark,
-			UiMode.NightNo => AppTheme.Light,
-			_ => AppTheme.Unspecified
-		};
+			var config = Application.Context.Resources?.Configuration;
+			if (config == null)
```
Comment (mandel-macaque) [inline src/Essentials/src/AppInfo/AppInfo.android.cs:48]:
```suggestion
			if (config is null)
```

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
minor changes.

File: src/Controls/tests/Core.UnitTests/MockPlatformServices.cs (hunk 1 above)
Comment (Redth) [reply src/Controls/tests/Core.UnitTests/MockPlatformServices.cs:89]: If we want these changes we should do a PR to main and not lose the changes in this backport PR. 

I don't think we need to block the backport PR on this, but perhaps we get a PR to main with the fixes before we complete this backport PR?

**Review by mandel-macaque — APPROVED** [review]
Approving based on the fact that it is a backport.

**Pull request merged**